### PR TITLE
Combining two forms into one, since they both for a new connection

### PR DESCRIPTION
**Pull request by n3op2**
# Pull Request

## Checklist
- [x] Have you read Digital Catapult's [Code of Conduct](https://github.com/digicatapult/.github/blob/main/CODE_OF_CONDUCT.md)?
- [x] I have performed a self-review of my own code.
- [x] I have commented my code, particularly in hard-to-understand areas.
- [x] I have made corresponding changes to the documentation.
- [x] My changes generate no new warnings.
- [ ] I have added tests that prove my fix is effective or that my feature works.
- [x] New and existing unit tests pass locally with my changes.

## PR Type

Please delete options that are irrelevant.

- [x] Code style update (formatting, local variables)

## Linked tickets

https://digicatapult.atlassian.net/jira/software/c/projects/ENG/boards/206?quickFilter=175&selectedIssue=VR-153&sprints=620

## High level description

Abstracted two form into one

## Detailed description

We used to have two different classes for each form e.g. sending invitation and verifying.

## Operational impact

None - as far as I'm aware

### CODE REVIEW

File: src/controllers/connection/__tests__/index.test.ts [hunk 1]
```diff
@@ -16,7 +16,7 @@ describe('ConnectionController', () => {
       let { dbMock, mockLogger, templateMock } = withConnectionMocks()
       const controller = new ConnectionController(dbMock, templateMock, mockLogger)
       const result = await controller.listConnections().then(toHTMLString)
-      expect(result).to.equal('list_foo-verified_list')
+      expect(result).to.equal('list_"foo"-verified_list')
```
Comment (Ellenn-A) [inline src/controllers/connection/__tests__/index.test.ts:19]: why is the foo in  "" now? (I just can't find where it came from)

Comment (n3op2) [reply src/controllers/connection/__tests__/index.test.ts:19]: This is from `JSON.stringify()` did not want to overdue with logic as this is mainly to increase assertation. A good attached a good example
Screenshot 2024-07-03 at 15 10 19

**Review by Ellenn-A — COMMENTED** [review]
looks good to me

File: src/views/newConnection/newInvite.tsx
Comment (mattdean-digicatapult) [inline src/views/newConnection/newInvite.tsx:1]: I don't see the value in this change sorry. There's little to no overlap between the two forms apart from what is already held in the base class which you've for some reason left in (even though now only one thing is derived from it). All this PR does is move code from one form to the other. 

Can I understand the motivation for amalgamating these forms into one beyond just you don't like the code structure?

**Review by mattdean-digicatapult — CHANGES_REQUESTED** [review]

**Comment by n3op2** [comment]
this is a duplicate, and also since this refactor is no longer relevant other commits have been pushed to this branch

**Pull request closed without merging**